### DISCUSSION TITLE
GSAP Animations

### DISCUSSION THREAD

**Discussion by memdigital** [elementor/elementor#27748] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I am frustrated that every other builder appears to be adding some really amazing animations and tricks utilising GSAP.

### Describe the solution you'd like.

It would be an amazing feature to have GSAP available within Elementor as standard.

To be able to control scroll to speeds, and smoothness. To add in reveals and scrolling animations.

### Describe alternatives you've considered.

There are plugins, or manual methods, all of which I have tried, but still would be amazing to see this included.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by sanholo177** [comment]
This has been most needed. While Wix, and Webflow have a really good integration with Gsap, Smooth Scroll, and Lenis. Out of the box they provide modern smooth animation controls and presets, while also giving opportunity to users to create more complex animations using different javascript libraries and data-controls.

On the other hand, it's very difficult to use Gsap even manually, let alone out-of-the-box animation. Elementor adds so many extra dom and classes that you can't target the text or images directly if you want to use GSAP to animate them. For example, using Split Type to break down the text into words and characters and animate them on scroll. Whilst on let's say Webflow if you add a custom class, it adds the class directly on the heading, on Elementor it adds on the Elementor Widget wrapper. 

We have over 100 websites with Elementor, and a lot of our clients ask us to create Awwwards-inspired websites with smooth scroll, GSAP reveal animations, locomotive scroll and we have to use a workaround all the time. It's simply not efficient and possible to replicate a website like that with Elementor. 

Webflow and Wix understand where Web Design is heading and they are actively shaping their platform toward that. There are advantages of Elementor, including a list of Widgets, but in terms of modern user experience, elementor lacks a lot. 

They have "**Motion Effects and Animation Updates** - Elevate your designs to create more engaging sites" on their /roadmap since last year. Hopefully, it arrives soon and its feature-packed, giving a space/opportunity to advanced developers to create more complex animation.

**Comment by coolj92** [comment]
@bhambrabland GSAP is now free for all! I agree that this would be a great addition to Elementor. https://gsap.com/pricing/ (free announcement).
